src/librustdoc/html/format.rs

```diff
@@ -460,7 +460,8 @@ impl fmt::Display for clean::Type {
                 f.write_str(name)
             }
             clean::ResolvedPath{ did, ref typarams, ref path } => {
-                try!(resolved_path(f, did, path, false));
+                // Paths like Self::Output should be rendered with all segments
```

Comment:
This is a good point... the ideal rendering of something like that in rustdoc would be `Iter<T>::Item`, with `Iter` hyperlinked to the appropriate place. So is it a matter of walking path segments and looking for the first non-module when printing them in "short" (less than fully-qualified) form?